File: projetos-fenix-para-estudo/t2ti-erp-fenix-banco-agencia-pessoa-completos/fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/service/cadastros/BancoAgenciaService.java
package com.t2ti.fenix.service.cadastros;

import java.util.List;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import javax.persistence.Query;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.t2ti.fenix.model.cadastros.BancoAgencia;
import com.t2ti.fenix.model.transiente.Filtro;
import com.t2ti.fenix.repository.cadastros.BancoAgenciaRepository;

@Service
public class BancoAgenciaService {

	@Autowired
	private BancoAgenciaRepository repository;
	
	@PersistenceContext
    private EntityManager entityManager;	
	
	public List<BancoAgencia> consultarLista() {
		return repository.findAll();
	}

	@SuppressWarnings("unchecked")
	public List<BancoAgencia> consultarLista(Filtro filtro) {
		String sql = "select * from banco_agencia where " + filtro.getWhere();
		Query query = entityManager.createNativeQuery(sql, BancoAgencia.class);
		List<BancoAgencia> lista = (List<BancoAgencia>) query.getResultList();
		return lista;
	}
	
	public BancoAgencia consultarObjeto(Integer id) {
		return repository.findById(id).get();
	}
	
	public BancoAgencia salvar(BancoAgencia objeto) {
		return repository.save(objeto);
	}
		
	public void excluir(Integer id) {
		BancoAgencia objeto = new BancoAgencia();
		objeto.setId(id);
		repository.delete(objeto);
	}
	
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-banco-agencia-pessoa-completos/fontes/backend/php/fenix/src/service/cadastros/BancoAgenciaService.php
<?php

class BancoAgenciaService extends ServiceBase
{
    public static function consultarLista()
    {
        $gerenteConexao = GerenteConexao::getInstance();        
        $objetoRepository = $gerenteConexao->entityManager->getRepository('BancoAgencia');
        return $objetoRepository->findAll();
    }

    public static function consultarListaFiltroValor(string $campo, string $valor)
    {
        $gerenteConexao = GerenteConexao::getInstance();   
        $sql = "SELECT t FROM BancoAgencia t WHERE t.$campo like :valor";     
        $query = $gerenteConexao->entityManager->createQuery($sql);
        $query->setParameter("valor", "%$valor%");
        return $query->getResult();
    }

    public static function consultarObjeto(int $id)
    {
        $gerenteConexao = GerenteConexao::getInstance();        
        return $gerenteConexao->entityManager->find('BancoAgencia', $id);
    }

    public static function salvar($objeto)
    {
        parent::salvarBase($objeto);
    }

    public static function excluir($objeto)
    {
        parent::excluirBase($objeto);
    }

}